### PR TITLE
Implement useAuthEmulator and remove auth.useEmulator.

### PR DESCRIPTION
**Pull request by yuchenshi**
Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

(Discussed internally)

### Testing

All tests are just moved to a new file and then I changed `auth.useEmulator(...)` to `useAuthEmulator(auth, ...)`. No changes in behavior.

### API Changes

This removes the `auth.useEmulator(...)` non-modular API and adds `useAuthEmulator(auth, ...)` modular API, as discussed internally. I'd appreciate we merge this PR ASAP to minimize impact. I can clean up the logic in a follow up PR.

### CODE REVIEW

File: packages-exp/auth-exp/src/core/auth/emulator.test.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import * as fetch from '../../../test/helpers/mock_fetch';
+import { Endpoint } from '../../api';
+import { User } from '../../model/user';
+import { _castAuth } from './auth_impl';
+import { useAuthEmulator } from './emulator';
+
+use(sinonChai);
+use(chaiAsPromised);
+
+describe('core/auth/emulator', () => {
```
Comment (yuchenshi) [inline packages-exp/auth-exp/src/core/auth/emulator.test.ts:36]: All tests moved from `auth_impl.ts`. No changes except API.

File: packages-exp/auth-exp/src/core/auth/emulator.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * @param options.disableWarnings - (Optional: default false) Disable the warning banner attached to the DOM
+ *
+ * @public
+ */
+export function useAuthEmulator(
+  auth: externs.Auth,
+  url: string,
+  options?: { disableWarnings: boolean }
+): void {
+  const authInternal = _castAuth(auth);
```
Comment (yuchenshi) [inline packages-exp/auth-exp/src/core/auth/emulator.ts:48]: Logic below moved from `auth_impl.ts` with `s/this/authInternal/g`.

File: packages-exp/auth-exp/src/core/auth/emulator.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    authInternal,
+    AuthErrorCode.INVALID_EMULATOR_SCHEME
+  );
+
+  authInternal.config.emulator = { url };
+  authInternal.settings.appVerificationDisabledForTesting = true;
+  emitEmulatorWarning(!!options?.disableWarnings);
+}
+
+function emitEmulatorWarning(disableBanner: boolean): void {
```
Comment (yuchenshi) [inline packages-exp/auth-exp/src/core/auth/emulator.ts:66]: Moved verbatim.

File: packages-exp/auth-exp/src/core/auth/emulator.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * @remarks
+ * This must be called synchronously immediately following the first call to
+ * {@link @firebase/auth#initializeAuth}.  Do not use with production credentials as emulator
+ * traffic is not encrypted.
+ *
+ *
+ * @example
+ * ```javascript
+ * useAuthEmulator(auth, 'http://127.0.0.1:9099');
```
Comment (itsagithubact) [inline packages-exp/auth-exp/src/core/auth/emulator.ts:34]: Wdyt about adding the disable warnings option to the example?

**Review by itsagithubact — APPROVED** [review]

File: packages-exp/auth-exp/src/core/auth/emulator.ts (hunk 4 above)
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/auth/emulator.ts:34]: Wdyt about adding disable warnings to the example?

**Review by samhorlbeck — APPROVED** [review]
Accidentally used my test account there before

File: packages-exp/auth-exp/src/core/auth/emulator.ts (hunk 4 above)
Comment (yuchenshi) [reply packages-exp/auth-exp/src/core/auth/emulator.ts]: Done.

**Review by samtstern — APPROVED** [review]

File: .changeset/nasty-lemons-bow.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+"@firebase/auth-compat": patch
```
Comment (hsubox76) [inline .changeset/nasty-lemons-bow.md:2]: No changeset needed for exp packages, this file can be removed.

Comment (yuchenshi) [reply .changeset/nasty-lemons-bow.md:2]:
```suggestion
```

File: .changeset/nasty-lemons-bow.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+"@firebase/auth-exp": patch
```
Comment (hsubox76) [inline .changeset/nasty-lemons-bow.md:2]: Sorry if I wasn't clear, the whole file can be removed.

Comment (yuchenshi) [reply .changeset/nasty-lemons-bow.md]: Gone!

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**